cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -452,144 +436,57 @@ func StartControllers(controllers map[string]InitFunc, s *options.CMServer, root
-			Cloud:                     cloud,
-			ClusterName:               s.ClusterName,
-			VolumeInformer:            sharedInformers.Core().V1().PersistentVolumes(),
-			ClaimInformer:             sharedInformers.Core().V1().PersistentVolumeClaims(),
-			ClassInformer:             sharedInformers.Storage().V1().StorageClasses(),
-			EnableDynamicProvisioning: s.VolumeConfiguration.EnableDynamicProvisioning,
+	var rootCA []byte
+	if ctx.Options.RootCAFile != "" {
+		rootCA, err = ioutil.ReadFile(ctx.Options.RootCAFile)
+		if err != nil {
+			return true, fmt.Errorf("error reading root-ca-file at %s: %v", ctx.Options.RootCAFile, err)
 		}
-		volumeController, volumeControllerErr := persistentvolumecontroller.NewController(params)
-		if volumeControllerErr != nil {
-			return fmt.Errorf("failed to construct persistentvolume controller: %v", volumeControllerErr)
+		if _, err := certutil.ParseCertsPEM(rootCA); err != nil {
+			return true, fmt.Errorf("error parsing root-ca-file at %s: %v", ctx.Options.RootCAFile, err)
 		}
-		go volumeController.Run(stop)
-		time.Sleep(wait.Jitter(s.ControllerStartInterval.Duration, ControllerStartJitter))
 	} else {
-		glog.Warningf("%q is disabled", pvBinderControllerName)
+		rootCA = c.rootClientBuilder.ConfigOrDie("tokens-controller").CAData
 	}
 
-	if ctx.IsControllerEnabled(attachDetachControllerName) {
-		if s.ReconcilerSyncLoopPeriod.Duration < time.Second {
-			return fmt.Errorf("Duration time must be greater than one second as set via command line option reconcile-sync-loop-period.")
-		}
-		attachDetachController, attachDetachControllerErr :=
-			attachdetach.NewAttachDetachController(
-				clientBuilder.ClientOrDie("attachdetach-controller"),
-				sharedInformers.Core().V1().Pods(),
-				sharedInformers.Core().V1().Nodes(),
-				sharedInformers.Core().V1().PersistentVolumeClaims(),
-				sharedInformers.Core().V1().PersistentVolumes(),
-				cloud,
-				ProbeAttachableVolumePlugins(s.VolumeConfiguration),
-				s.DisableAttachDetachReconcilerSync,
-				s.ReconcilerSyncLoopPeriod.Duration)
-		if attachDetachControllerErr != nil {
-			return fmt.Errorf("failed to start attach/detach controller: %v", attachDetachControllerErr)
-		}
-		go attachDetachController.Run(stop)
-		time.Sleep(wait.Jitter(s.ControllerStartInterval.Duration, ControllerStartJitter))
-	} else {
-		glog.Warningf("%q is disabled", attachDetachControllerName)
-	}
+	controller := serviceaccountcontroller.NewTokensController(
+		ctx.InformerFactory.Core().V1().ServiceAccounts(),
+		ctx.InformerFactory.Core().V1().Secrets(),
+		c.rootClientBuilder.ClientOrDie("tokens-controller"),
+		serviceaccountcontroller.TokensControllerOptions{
+			TokenGenerator: serviceaccount.JWTTokenGenerator(privateKey),
+			RootCA:         rootCA,
+		},
+	)
+	go controller.Run(int(ctx.Options.ConcurrentSATokenSyncs), ctx.Stop)
 
-	sharedInformers.Start(stop)
+	// start the first set of informers now so that other controllers can start
+	ctx.InformerFactory.Start(ctx.Stop)
```

Comment:
is it odd to tie sharedInformers to this controller? if you're not running with `--use-service-account-credentials`, what else in the controller manager depends on this controller?